### DISCUSSION TITLE
Audio Mic Input -> I2S -> SD Card

### DISCUSSION THREAD

**Discussion by stumpie24** [pschatzmann/arduino-audio-tools#877] [Q&A] [answered]
There is a way to accepting an SD card file to the microprocessor. However, I do not believe there is a written file that takes an input from a microphone and stores the input data to an SD card. In terms of save format, .wav would be best if you want stored a wider variety of data, but an mp3 would work from terms of wanting a smaller size file stored on the SD card.

**Comment by pschatzmann** [comment, marked as answer]
There is all the functionality you need! Files stored on the SD are just subclasses of Arduino Streams so you can use them as copy destination  

If you want to use some other format then PCM you will need to read the tutorial how to use the codecs. MP3 is asking for a lot of memory and cpu resources, so I would suggest that you try this only if you know what you are doing. There are codecs that are better suited for microcontrollers. See https://www.pschatzmann.ch/home/2023/05/10/fast-audio-codecs-for-microcontrollers/
